### PR TITLE
fix app meta tags (fixes #770; related to #316)

### LINKED ISSUE #770: Do not inject `apple-mobile-web-app-capable` and `apple-mobile-web-app-status-bar-style` `<meta>` tags if they already exist
<a href="https://github.com/cvan"></img></a> **Issue by [cvan](https://github.com/cvan)**
_Friday Dec 18, 2015 at 02:56 GMT_
_Originally opened as https://github.com/aframevr/aframe-core/issues/709_

----

We [check for `meta[name="viewport"]`](https://github.com/aframevr/aframe-core/blob/2603f4e/src/core/a-scene.js#L795-L796) before [we create and inject one](https://github.com/aframevr/aframe-core/blob/2603f4e/src/core/a-scene.js#L803), but we are creating and injecting [`meta[name="apple-mobile-web-app-capable"]`](https://github.com/aframevr/aframe-core/blob/2603f4e/src/core/a-scene.js#L806-L809) and [`meta[name="apple-mobile-web-app-status-bar-style"]`](https://github.com/aframevr/aframe-core/blob/2603f4e/src/core/a-scene.js#L811-L814) tags even if they already exist.

### PR DESCRIPTION
**Pull request by cvan**

### CODE REVIEW

File: tests/core/a-scene.test.js [hunk 1]
```diff
@@ -201,8 +207,27 @@ helpers.getSkipCISuite()('a-scene (with renderer)', function () {
...
-      assert.ok(document.querySelector('meta[name="viewport"]'));
+    test('sets up meta tags (non-iOS)', function () {
+      assert.equal(document.querySelector('meta[name="viewport"]').content,
+        'width=device-width,initial-scale=1,shrink-to-fit=no,user-scalable=no');
+      assert.ok(document.querySelector('meta[name="web-app-capable"]'));
+      assert.notOk(document.querySelector('meta[name="apple-mobile-web-app-capable"]'));
+    });
+
+    test('sets up meta tags (iOS)', function (done) {
+      var el = this.el;
```
Comment (cvan) [inline tests/core/a-scene.test.js]: what's a better way of patching this?

File: docs/primitives/a-video.md [hunk 2]
```diff
@@ -37,6 +37,6 @@ Note that the video primitive inherits common [mesh attributes](./mesh-attribute
 
 iOS has a lot of restrictions on playing videos in the browser. To play an inline video texture, we must:
 
-- Set the `<meta name="apple-mobile-web-app-capable" content="yes"` metatag.
+- Set the `<meta name="apple-mobile-web-app-capable" content="yes">` meta tag (will be inject if missing).
```
Comment (ngokevin) [inline docs/primitives/a-video.md]: injected

File: tests/core/a-scene.test.js [hunk 3]
```diff
@@ -188,11 +188,17 @@ helpers.getSkipCISuite()('a-scene (with renderer)', function () {
   setup(function (done) {
     var el;
     var self = this;
+    // Remove meta tag in Karma page.
```
Comment (ngokevin) [inline tests/core/a-scene.test.js]: Might want to move the metatag tests to a separate suite at this point

File: docs/primitives/a-video.md (hunk 2 above)
Comment (cvan) [inline docs/primitives/a-video.md]: good catch

File: tests/core/a-scene.test.js (hunk 1 above)
Comment (ngokevin) [inline tests/core/a-scene.test.js]: different test suite with your own setup

File: src/utils/index.js [hunk 4]
```diff
@@ -125,7 +125,7 @@ module.exports.isMobile = function () {
 };
 
 var isIOS = module.exports.isIOS = function () {
-  return /iPad|iPhone|iPod/.test(navigator.platform);
+  return /i(pad|phone|pod)/i.test(navigator.platform);
```
Comment (dmarcos) [inline src/utils/index.js]: Do you think this makes the regular expression easier to understand?

File: tests/core/a-scene.test.js (hunk 3 above)
Comment (dmarcos) [inline tests/core/a-scene.test.js]: Yeah the scene tests are getting pretty big.

File: src/utils/index.js (hunk 4 above)
Comment (cvan) [inline src/utils/index.js]: Yeah lol

File: tests/core/a-scene.test.js (hunk 1 above)
Comment (cvan) [inline tests/core/a-scene.test.js]: Figured - ok I'll do that

File: src/utils/index.js (hunk 4 above)
Comment (dmarcos) [inline src/utils/index.js]: I think before it was more clear.

Comment (cvan) [inline src/utils/index.js]: Yeah just joshing. Will fix.

**Comment by dmarcos** [comment]
Is there intent to finish this up?

**Comment by cvan** [comment]
yes, was waiting for help from Kevin on the tests. otherwise, this works - and fixes a bug.

**Comment by cvan** [comment]
but I'll get to it. I'm not incredibly familiar with our tests, but I can figure it out no issue.

**Comment by cvan** [comment]
tests fixed and added

File: src/core/scene/metaTags.js [hunk 5]
```diff
@@ -1,46 +1,78 @@
-module.exports = function initMetaTags (scene) {
-  if (!scene.isMobile) { return; }
-  injectMetaTags();
-};
+var extend = require('../../utils').extend;
+
+var MOBILE_HEAD_TAGS = module.exports.MOBILE_HEAD_TAGS = [
+  Meta({name: 'viewport', content: 'width=device-width,initial-scale=1,maximum-scale=1,shrink-to-fit=no,user-scalable=no'}),
```
Comment (dmarcos) [inline src/core/scene/metaTags.js]: Just to make sure. Does this PR consider the viewport problem with the twitter appview?

File: src/utils/index.js (hunk 4 above)
Comment (dmarcos) [inline src/utils/index.js]: pending change

File: src/core/scene/metaTags.js (hunk 5 above)
Comment (cvan) [inline src/core/scene/metaTags.js]: yep; just moving the code: `maximum-scale=1`

**Comment by dmarcos** [comment]
This blocks 0.2.0 release. We need to get it done.

**Comment by cvan** [comment]
> This blocks 0.2.0 release. We need to get it done.

this was completely ready yesterday. r+?

File: src/utils/index.js (hunk 4 above)
Comment (dmarcos) [inline src/utils/index.js]: I still think that the explicit list of words is more clear than a fancy regular expressions. Each time I have to do any regular expression I have to go and read a tutorial.

File: tests/core/a-scene.test.js [hunk 6]
```diff
@@ -124,3 +121,73 @@ helpers.getSkipCISuite()('a-scene (with renderer)', function () {
     sinon.assert.calledWith(Component.tick, scene.time);
   });
 });
+
+helpers.getSkipCISuite()('a-scene (for iOS)', function () {
```
Comment (ngokevin) [inline tests/core/a-scene.test.js]: I can help with this in a follow-up patch, but I don't think we need rendering for these tests, and they can move to a `metaTags.test.js` suite.

File: src/core/scene/a-scene.js [hunk 7]
```diff
@@ -12,7 +12,8 @@ var AEntity = require('../a-entity');
 var ANode = require('../a-node');
 
 var registerElement = re.registerElement;
-var isMobile = utils.isMobile();
+var isIOS = utils.isMobile();
```
Comment (ngokevin) [inline src/core/scene/a-scene.js]: this looks backwards

**Comment by ngokevin** [comment]
Tagged need changes for the  iOS = mobile / mobile = iOS, not the tests

File: src/core/scene/a-scene.js (hunk 7 above)
Comment (dmarcos) [inline src/core/scene/a-scene.js]: Yeah these two guys look awkward. :)

Comment (cvan) [inline src/core/scene/a-scene.js]: oops! good call

File: tests/core/a-scene.test.js (hunk 6 above)
Comment (cvan) [inline tests/core/a-scene.test.js]: yeah, that probably makes more sense now that I think about it

**Comment by cvan** [comment]
thanks, good catches

**Comment by ngokevin** [comment]
Code looks good to me, I will follow-up to help with tests.

**Comment by cvan** [comment]
> Code looks good to me, I will follow-up to help with tests.

thanks, man, much appreciated :+1:

**Pull request merged**